test/SILGen/default_arguments.swift

```diff
@@ -446,11 +446,11 @@ enum E {
   // CHECK-LABEL: sil hidden [ossa] @$s17default_arguments1EO4testyyFZ : $@convention(method) (@thin E.Type) -> ()
   static func test() {
     // CHECK: function_ref @$s17default_arguments1EO6ResultV4name9platformsAESS_SaySiGtcfcfA0_ : $@convention(thin) () -> @owned Array<Int>
-    // CHECK: function_ref @$s17default_arguments1EO4testyyFZAC6ResultVSS_SaySiGtcfu_ : $@convention(thin) (@guaranteed String, @guaranteed Array<Int>) -> @owned E.Result
+    // CHECK: function_ref @$s17default_arguments1EO6ResultV4name9platformsAESS_SaySiGtcfC : $@convention(method) (@owned String, @owned Array<Int>, @thin E.Result.Type) -> @owned E.Result
 
-    // CHECK-LABEL: sil private [ossa] @$s17default_arguments1EO4testyyFZAC6ResultVSS_SaySiGtcfu_ : $@convention(thin) (@guaranteed String, @guaranteed Array<Int>) -> @owned E.Result
     var result = Self.Result(name: "")
   }
+  // CHECK: end sil function '$s17default_arguments1EO4testyyFZ'
 }
```

Comment:
Minus a thunk here because `Self.Result` is now folded into a type expression. It used to not because `TypeExpr::createForMemberDecl` accepted only a `DeclRefTypeRepr` parent, and `Self` is modeled as a `FixedTypeRepr`, not an identifier type representation.